packages/dev-server-hmr/src/HmrPlugin.ts

```diff
@@ -0,0 +1,284 @@
+import type {
+  Plugin,
+  WebSocketsManager,
+  Logger,
+  WebSocketData,
+  ServerStartParams,
+} from '@web/dev-server-core';
+import { appendHtmlToDocument } from '@web/dev-server-core';
+import WebSocket from 'ws';
+import type { Context } from 'koa';
+import { hmrClientScript } from './hmrClientScript';
+import { posix as pathUtil } from 'path';
+
+export interface HmrReloadMessage {
+  type: 'reload';
+}
+
+export interface HmrUpdateMessage {
+  type: 'update';
+  url: string;
+}
+
+export interface HmrAcceptMessage extends WebSocketData {
+  type: 'hotAccept';
+  id: string;
+}
+
+export type HmrMessage = HmrReloadMessage | HmrUpdateMessage;
+
+export type HmrClientMessage = HmrAcceptMessage;
+
+export interface HmrModule {
+  dependencies: Set<string>;
+  dependents: Set<string>;
+  hmrAccepted: boolean;
+  hmrEnabled: boolean;
+}
+
+export const NAME_HMR_CLIENT_IMPORT = '/__web-dev-server__hmr.js';
+
+export const hmrClientImport = `<!-- injected by web-dev-server -->
+<script type="module" src="${NAME_HMR_CLIENT_IMPORT}"></script>`;
```

Comment:
Why do we inject this in the HTML page here. Doesn't it already get injected into the modules that need it?

Reply:
leftover code from my first attempt, good catch